app/hotels/src/HotelsContext.js

```diff
@@ -118,19 +118,21 @@ export default { Consumer, Provider };
 
 export function withHotelsContext(select: (state: State) => Object) {
   return function(Component: React.ElementType) {
-    const WithHotelsContext = (props: Object) => {
-      const mapStateToProps = state => {
+    class WithHotelsContext extends React.Component<Object> {
+      // $FlowExpectedError: We need to pass on the navigationOptions if any, flow does not know about it, but a react component might have it
+      static navigationOptions = Component.navigationOptions;
+
+      mapStateToProps = (state: State) => {
         const stateProps = select(state);
-        return <Component {...props} {...stateProps} />;
+        return <Component {...this.props} {...stateProps} />;
       };
 
-      return <Consumer>{mapStateToProps}</Consumer>;
-    };
-
-    // $FlowExpectedError: We need to pass on the navigationOptions if any, flow does not know about it, but a react component might have it
-    if (Component.navigationOptions) {
-      WithHotelsContext.navigationOptions = Component.navigationOptions;
+      render() {
+        return <Consumer>{this.mapStateToProps}</Consumer>;
+      }
     }
+    // $FlowExpectedError: We need to pass on the navigationOptions if any, flow does not know about it, but a react component might have it
```

Comment:
This should probably be removed.

Reply:
Did you try to remove it? I think it is still needed 😊

Reply:
![screen shot 2018-10-31 at 09 00 22](https://user-images.githubusercontent.com/22741774/47773771-71d4d900-dceb-11e8-8926-28b54b971f49.png)
No error it seems.

Reply:
Oh, sorry, I missed that there was one below as well, this can obviously be removed, good catch.